src/events/websocket/WebSocketClients.js

```diff
@@ -82,49 +86,108 @@ export default class WebSocketClients {
+          }
+
+          if (!functionKey) {
+            return
+          }
+
+          const sendError = (err) => {
+            let message
+            if (route === '$connect') {
+              message = err.message
+            } else {
+              message = stringify({
+                connectionId,
+                message: 'Internal server error',
+                requestId: '1234567890',
+              })
+            }
+
+            if (websocketClient.readyState === OPEN) {
+              websocketClient.send(message)
+            }
+
+            // mimic AWS behaviour (close connection) when the $connect route handler throws
+            if (route === '$connect') {
+              websocketClient.close()
+            }
+
+            debugLog(`Error in route handler '${functionKey}'`, err)
+          }
+
+          // let result
+          const authorizedEvent = event
+
+          try {
+            let authorization = this.#clientAuthorization.get(connectionId)
+            const lambdaAuthorizer = this.#webSocketAuthorizers.get(functionKey)
+
+            if (lambdaAuthorizer && !authorization && route !== '$disconnect') {
+              if (lambdaAuthorizer) {
+                const lambdaAuthorizerFunction = this.#lambda.get(
+                  lambdaAuthorizer.name.toLowerCase(),
+                )
+                lambdaAuthorizerFunction.setEvent(authorizedEvent)
+                authorization = await lambdaAuthorizerFunction.runHandler()
+                this.#clientAuthorization.set(connectionId, authorization)
+              }
+            }
+
+            let authorizerFail = true
+
+            if (lambdaAuthorizer) {
+              if (authorization) {
+                const policy = authorization.policyDocument
+                if (
+                  policy &&
+                  policy.Statement &&
+                  policy.Statement.length === 1
+                ) {
+                  const statement = policy.Statement[0]
+                  if (statement.Action === 'execute-api:Invoke') {
+                    if (statement.Effect === 'allow') {
+                      authorizedEvent.requestContext.authorizer =
+                        authorization.context
+                      authorizedEvent.requestContext.authorizer.principalId =
+                        authorization.principalId
+                      authorizerFail = false
+                    }
+                  }
+                }
+              }
+            } else {
+              authorizerFail = false
+            }
+
+            if (authorizerFail)
+              throw Error(
+                'HTTP Authentication failed; no valid credentials available',
+              )
+
+            const lambdaFunction = this.#lambda.get(functionKey)
+
+            lambdaFunction.setEvent(authorizedEvent)
+
+            /* result = */ await lambdaFunction.runHandler()
+
+            // TODO what to do with "result"?
```

Comment:
Is that supposed to be addressed as a part of this PR?

Reply:
Sorry, I don't understand your comment. These referenced lines 178-181 is not part of my code.